manifests/repo.pp

```diff
@@ -1,32 +1,53 @@
-# Configure package repository
+# @api private
+# @summary Configure package repository
 #
-class php::repo {
-  $msg_no_repo = "No repo available for ${facts['os']['family']}/${facts['os']['name']}"
+# @param [Boolean] manage_external_repo
+#   manage an external package repository or not
+# @option external_repo [Boolean] :false
+#   default
+# @option external_repo [Boolean] :true  
+#
+# @param [Array] external_repo_supported_php_versions
+#   The PHP versions the additional repositories support
+# @option external_repo_supported_php_versions [Array] :[]
+#   default: empty array, default for OS from Hiera Data in module
+#
+# @param [Boolean] manage_internal_repo
+#   manage an internal package repository, parameter is to control
+#   app streams in CENTOS/RHEL/Fedora, Repos on FreeBSD or Archlinux
+# @option external_repo [Boolean] :false
+#   true
+# @option external_repo [Boolean] :true   
+#
+# @param [Array] os_supported_php_versions
+#   The PHP versions the OS supports by default, without any additional repositories,
+#   on RHEL/CentOS and Fedora this includes app streams, on FreeBSD or Archlinux it manages Repos
+# @option os_supported_php_versions [Array] :[]
+#   default: empty array, default for OS from Hiera Data in module
+# 
+# @param [String] php_version
+#   Set the PHP version to install
+# @option php_version [String] :'7.4'
+#   defaults: latest stable PHP Version available on php.net, default for OS from Hiera Data in module
+#
+class php::repo (
+  Boolean $manage_internal_repo                         = $php::manage_internal_repo,
+  Boolean $manage_external_repo                         = $php::manage_external_repo,
+  Optional[Array] $external_repo_supported_php_versions = $php::external_repo_supported_php_versions,
+  Array $os_supported_php_versions                      = $php::os_supported_php_versions,
+  Optional[String[1]] $php_version                      = $php::php_version,
 
-  case $facts['os']['family'] {
-    'Debian': {
-      # no contain here because apt does that already
-      case $facts['os']['name'] {
-        'Debian': {
-          include php::repo::debian
-        }
-        'Ubuntu': {
-          include php::repo::ubuntu
-        }
-        default: {
-          fail($msg_no_repo)
-        }
-      }
-    }
-    'FreeBSD': {}
-    'Suse': {
-      contain php::repo::suse
-    }
-    'RedHat': {
-      contain 'php::repo::redhat'
-    }
-    default: {
-      fail($msg_no_repo)
+) {
+  assert_private()
+
+  if (($manage_internal_repo or $manage_external_repo) and (($php_version in $external_repo_supported_php_versions) or ($php_version in $os_supported_php_versions))) {
+    case $facts['os']['name'] {
+      'Archlinux':                  { contain 'php::repo::archlinux' }
```

Comment:
for arch and freebsd, should we just do nothing? since those don't have external repositories.
```suggestion
      'Archlinux':                  { }
```

Reply:
The classes are for generally managing repos, is there no need to connect a specific one, for a specific php version? Kind of the same problem exists for RHEL style distros, there is the need to activate a specific app stream

Reply:
Setz ich so um